python/pyarrow/table.pxi

```diff
@@ -2009,6 +2009,24 @@ cdef class Table(_PandasConvertible):
             result.append(pyarrow_wrap_batch(batch))
 
         return result
+    
+    def to_reader(self):
```

Comment:
Add a quick unit test? Also, add the `max_chunksize` argument like `to_batches`?

Reply:
Also, we should add a quick C++ unit test as well (where we make sure to drop the table or move the table into the reader before using the reader)